beacon_node/beacon_chain/tests/store_tests.rs

```diff
@@ -2599,17 +2602,18 @@ async fn weak_subjectivity_sync_test(slots: Vec<Slot>, checkpoint_slot: Slot) {
                 .clone()
                 .deconstruct();
             if wss_fork.fulu_enabled() {
-                let (data_columns, expected_column_indices) = cols.unwrap();
+                info!(block_slot = %block.slot(), ?block_root, "Corrupting data column KZG proof");
+                let (mut data_columns, expected_column_indices) = cols.unwrap();
                 assert!(
                     !data_columns.is_empty(),
                     "data column sidecars shouldn't be empty"
                 );
-                let mut sidecar = data_columns[0].clone_arc();
-                let mut_sidecar = Arc::make_mut(&mut sidecar);
-                if mut_sidecar.kzg_proofs[0] == KzgProof::empty() {
+                let mut data_column = (*(data_columns[0]).clone_arc()).clone();
+                if data_column.kzg_proofs[0] == KzgProof::empty() {
                     panic!("kzg_proof is already G1_POINT_AT_INFINITY")
                 }
-                mut_sidecar.kzg_proofs[0] = KzgProof::empty();
+                data_column.kzg_proofs[0] = KzgProof::empty();
+                data_columns[0] = CustodyDataColumn::from_asserted_custody(data_column.into());
```

Comment:
Changed `Arc::make_mut` into a direct assignment to make sure we are mutating the original Vec